### DISCUSSION TITLE
Target *Explicit* Hyperlinks without the mouse?

### DISCUSSION THREAD

**Discussion by chrisgrieser** [wezterm/wezterm#4349] [Q&A]
### What Operating System(s) are you running on?

macOS

### WezTerm version

wezterm 20230712-072601-f4abf8fd

### Ask your question!

So with the [quick select feature and a custom action](https://wezfurlong.org/wezterm/config/lua/keyassignment/QuickSelectArgs.html), it is easily possible to open a link with the keyboard.

However, as far as I can tell, that only works for implicit hyperlinks, not [explicit hyperlinks](https://wezfurlong.org/wezterm/hyperlinks#explicit-hyperlinks). I tried using things `pattern = { [[\e]8;;.*]] }` to specifically target explicit hyperlinks, not without success. 

Can *explicit* hyperlinks really only be opened with the mouse or am I missing something here?

**Comment by wez** [comment]
quickselect and search functions operate on the textual portion of the display.  Hyperlinks are attributes and are not able to be matched that way.

I've been thinking about adding a parameter to https://wezfurlong.org/wezterm/config/lua/keyassignment/QuickSelectArgs.html that would match all spans of text that have hyperlinks set, so you'd either match the text, or match the hyperlink attribute.

**Reply by ghost** [reply]
Yes, that would be awesome!
Hyperlinks are something to be interacted with, so making it possible to open those hyperlinks via your keyboard with the QuickSelect only makes sense to me.
